dev/AnimatedIcon/AnimatedIconDevDesign.md

```diff
@@ -0,0 +1,135 @@
+# AnimatedIcon Dev Design
+This doc details the technical challenges and specifications of the animated icon control.
+
+## IDL Definition
+```
+Interface IRichAnimatedVisualSource : IAnimatedVisualSource
+{
+    IAnimatedVisual TryCreateAnimatedVisual(Windows.UI.Compositor compositor, out object diagnostics);
+    Windows.Foundation.Collections.IMapView<String, Double> Markers { get; };
+    void SetColorProperty(String propertyName, Windows.UI.Color value);
+}
+
+public class StatefulIcon : IconElement  
+{ 
+    [MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnStatePropertyChanged")]
+    String StateGroup1{get; set;} //CommonStates
+    
+    [MUX_PROPERTY_CHANGED_CALLBACK_METHODNAME("OnStatePropertyChanged")]
+    String StateGroup2{get; set;} //DisabledStates
```

Comment:
Handling orthogonal state groups seems to drastically increase the complexity due to the combinations. We can start with a single group and see where we hit roadblocks with that. Adding more state groups should be an additive change.